code/modules/mob/living/living_say.dm

```diff
@@ -232,8 +232,8 @@ GLOBAL_LIST_INIT(message_modes_stat_limits, list(
 		spans |= SPAN_SINGING
 
 	// Leaving this here so that anything that handles speech this way will be able to have spans affecting it and all that.
-	var/sigreturn = SEND_SIGNAL(src, COMSIG_MOB_SAY, args, message_range)
-	if (sigreturn & COMPONENT_UPPERCASE_SPEECH)
+	var/sigreturn = SEND_SIGNAL(src, COMSIG_MOB_SAY, args + message_range + saymode)
```

Comment:
Do you want me to just add `message_range` and `saymode` to the args for `say()`?

Reply:
So I initially tried to do `SEND_SIGNAL(src, COMSIG_MOB_SAY, args, message_range, saymode)` but it kept saying the list index was out of bounds when I tried to access `SPEECH_SAYMODE`  I don't believe `SEND_SIGNAL` was merging the latter arguments correctly in the marco:

`#define SEND_SIGNAL(target, sigtype, arguments...) ( !target.comp_lookup?[sigtype] ? NONE : target._SendSignal(sigtype, list(target, ##arguments)) )`

When I was testing `SPEECH_RANGE` before my changes, it wouldn't showup in the args list.  Not to mention that `SPEECH_FILTERPROOF` was skipped accidentally.  CC @GoldenAlpharex 

The only place that touches SPEECH_RANGE is dullhans, so this might fix say code involving them.

Reply:
Also I dumped half a day trying to fix this minor bug that I thought was going to be easy.  Say code is torture...

Reply:
Let me know if https://github.com/tgstation/tgstation/pull/69844/commits/7144565b8161a1755cfac833ee38650553da7562 is fine.  I did some light testing with xenos, radios, and hand sign language with no problems.